### PR TITLE
feat: DeviceBluetooth - common

### PR DESCRIPTION
**Pull request by pkar70**
GitHub Issue (If applicable): part of #3045

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
No implementation of Bluetooth.

## What is the new behavior?
Common structs/enums are implemented for all platforms.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
 To be able to implement Windows.Devices.Enumeration.DeviceInformation.FindAllAsync for Bluetooth, various  (BluetoothDevice|BluetoothLEDevice).GetDeviceSelector methods are required.  Some of them requires another level of enums/structs (e.g. GetDeviceSelectorFromConnectionStatus() requires BluetoothConnectionStatus).

All selectors are copied from native UWP app.

Now, I have to dig into current implementation of DeviceInformation.FindAllAsync, that is used probably only for MIDI devices, and extend it to Bluetooth.
And there is much work to do, as e.g. AQS filter for MIDI uses System.Devices.InterfaceClassGuid , but for Bluetooth uses System.Devices.DevObjectType:=5 AND System.Devices.Aep.ProtocolId:="{E0CBF06C-CD8B-4647-BB8A-263B43F0F974}", so Uno.Devices.Enumeration.Internal.DeviceClassGuids shouldn't be DeviceClassGuids , etc.

Internal Issue (If applicable):

### CODE REVIEW

**Comment by pkar70** [comment]
Which is current place in diffignore to add two removed methods?
Or, better, can diffignore contain only data for "for current" build? and previous builds be in "archive" versions of diffignore?
And, again better, can diffignore be merged from small, per PR diffignore? It would simplify making PRs, as constant "resolve conflict" is not nice thing to do.

**Comment by pkar70** [comment]
Could this be reviewed/accepted?
Then I could make implementation of Bluetooth...

**Comment by pkar70** [comment]
Could this be reviewed/accepted? These are only structs/enums...
Then I could make implementation of Bluetooth.

**Comment by pkar70** [comment]
Can this PR be accepted? It is simple struct..

File: src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿using System;
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs:1]: Please use the default formatter for this file, the spacing and indentation is off.

File: src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			return _deviceSelectorPrefix + "(System.ItemNameDisplay:=\"" + deviceName + "\" OR " + _deviceSelectorIssueInquiry + "#True)";
+		}
+
+		public static string GetDeviceSelectorFromBluetoothAddress( ulong bluetoothAddress)
+		{ 
+			string macAddr = string.Format("{0:x12}", bluetoothAddress);
+			return _deviceSelectorPrefix + "(System.DeviceInterface.Bluetooth.DeviceAddress:=\"" + macAddr + "\" OR " + _deviceSelectorIssueInquiry + "#True)";
+		}
+
+		//method public static string GetDeviceSelectorFromClassOfDevice( BluetoothClassOfDevice classOfDevice)
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs:52]: Please remove

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		public static string GetDeviceSelectorFromDeviceName( string deviceName)
+		{
+			return _deviceSelectorPrefix + "(System.ItemNameDisplay:=\"" + deviceName + "\" OR " + _deviceSelectorIssueInquiry + "#True)";
+		}
+
+		public static string GetDeviceSelectorFromBluetoothAddress( ulong bluetoothAddress)
+		{
+			string macAddr = string.Format("{0:x12}", bluetoothAddress);
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:48]: Please do that the check notice mentioned.

File: src/Uno.UWP/Uno.csproj [hunk 4]
```diff
@@ -40,6 +40,19 @@
...
+	  <UpToDateCheckInput Remove="Devices\Bluetooth\BluetoothDevice.cs" />
+	  <UpToDateCheckInput Remove="Devices\Bluetooth\BluetoothLEAppearance.cs" />
+	  <UpToDateCheckInput Remove="Devices\Bluetooth\BluetoothLEDevice.cs" />
+	  <UpToDateCheckInput Remove="Devices\Enumeration\BluetoothClassOfDevice.cs" />
+	  <UpToDateCheckInput Remove="Devices\Enumeration\BluetoothConnectionStatus.cs" />
+	  <UpToDateCheckInput Remove="Devices\Enumeration\BluetoothMajorClass.cs" />
+	  <UpToDateCheckInput Remove="Devices\Enumeration\BluetoothMinorClass.cs" />
+	  <UpToDateCheckInput Remove="Devices\Enumeration\BluetoothServiceCapabilities.cs" />
+	  <UpToDateCheckInput Remove="Devices\Enumeration\DeviceInformationKind.cs" />
+	</ItemGroup>
```
Comment (jeromelaban) [inline src/Uno.UWP/Uno.csproj:55]: Please remove

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		public static string GetDeviceSelectorFromBluetoothAddress( ulong bluetoothAddress,  BluetoothAddressType bluetoothAddressType)
+		{
+			if(bluetoothAddressType == BluetoothAddressType.Unspecified)
+			{
+				return GetDeviceSelectorFromBluetoothAddress(bluetoothAddress);
+			}
+
+			string macAddr = string.Format("{0:x12}", bluetoothAddress);
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:59]: provide a culture info

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Windows.Devices.Bluetooth
+{
+	public partial class BluetoothLEDevice : global::System.IDisposable
+	{
+		private static string _deviceSelectorPrefix = "System.Devices.DevObjectType:=5 AND System.Devices.Aep.ProtocolId:=\"{BB7BB05E-5972-42B5-94FC-76EAA7084D49}\" AND ";
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:9]: Are the constant strings for this class coming from the execution on Windows?

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+			return _deviceSelectorPrefix + "(System.ItemNameDisplay:=\"" + deviceName + "\" OR " + _deviceSelectorIssueInquiry + "#True)";
+		}
+
+		public static string GetDeviceSelectorFromBluetoothAddress( ulong bluetoothAddress)
+		{
+			string macAddr = string.Format("{0:x12}", bluetoothAddress);
+			return _deviceSelectorPrefix + "(System.DeviceInterface.Bluetooth.DeviceAddress:=\"" + macAddr + "\" OR " + _deviceSelectorIssueInquiry + "#True)";
+		}
+
+		public static string GetDeviceSelectorFromBluetoothAddress( ulong bluetoothAddress,  BluetoothAddressType bluetoothAddressType)
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:52]: You'll need to add unit tests for this class, the logic has to be validated from regressions

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs (hunk 3 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:48]: Sorry, I don't understand. Can you rephrase this?

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs (hunk 6 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:9]: Yes, all selectors are from standard UWP - I created simple app to get all possible BT selectors, and copied these selectors.

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs (hunk 7 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:52]: Unit test - to check if selectors are same as in this PR implementation?

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:48]: I updated my comment.

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs (hunk 7 above)
Comment (jeromelaban) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:52]: Yes.

File: src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs (hunk 2 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs:52]: It takes some time to dig into working of this method; if someone wants to implement this method, he/she can use this info and save about hour... Really should I delete this?

File: src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs (hunk 1 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs:1]: ^K D applied (sequence found on Internet)

File: src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs (hunk 7 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothLEDevice.cs:52]: Ok, mostly done - not every selector, as it would be many pages long test (for every BLE.Appearance etc.).

File: src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs (hunk 2 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs]: Can I leave this comment in code?

Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs]: Can I leave this comment in code? if someone wants to implement this method, he/she can use this info and save some time (results of checks how this method work on UWP is described).

Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs]: Can I get answer to this question?

Comment (jeromelaban) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs]: Yes, please remove this. If this is useful information, please add this to a documentation file in the `features` directory

**Review by jeromelaban — COMMENTED** [review]
Please add a sample page that demonstrates the use of these APIs.

File: src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs (hunk 2 above)
Comment (pkar70) [reply src/Uno.UWP/Devices/Bluetooth/BluetoothDevice.cs]: Ok, I will move this info (how to implement this method) to new `features` page.

**Comment by pkar70** [comment]
Ok, I will add sample app. Pity I cannot copy it from PR with MIDI support :)

**Comment by pkar70** [comment]
Any more comments/requests for this PR? Or, is it ready to be merged?

**Comment by jeromelaban** [comment]
I ran the sample app, and this is what shows up, for reference:

image
